Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose
you have collected data on the amount of time students spend studying for an exam and their final exam
scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

In [3]:
import numpy as np
studying_time = [10, 8, 12, 9, 7]
exam_scores = [80, 75, 90, 85, 70]
pearsoncoef=np.corrcoef(studying_time,exam_scores)[0,1]
print(pearsoncoef)

0.9041944301794651


Pearson correlation coefficient is close to 1, it indicates a strong positive linear relationship. It means that as the amount of time spent studying increases, the final exam scores tend to increase as well.

Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables.
Suppose you have collected data on the amount of sleep individuals get each night and their overall job
satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two
variables and interpret the result.

In [12]:
import scipy.stats as stats

# Create the example dataset
sleep_amount = [7, 6, 8, 5, 7]
job_satisfaction = [9, 7, 8, 6, 9]

# Calculate the Spearman's rank correlation coefficient
spearman_coefficient, _ = stats.spearmanr(sleep_amount, job_satisfaction)

# Print the Spearman's rank correlation coefficient
print(spearman_coefficient)

0.6842105263157895


In [18]:
import pandas as pd
#import seaborn as sns
df=pd.DataFrame({'sleep_amount': [7, 6, 8, 5, 7],
                'job_satisfaction': [9, 7, 8, 6, 9]})

In [22]:
df.corr()

sleep_amount  job_satisfaction
sleep_amount          1.000000          0.773574
job_satisfaction      0.773574          1.000000

In [21]:
df.corr(method='spearman')

sleep_amount  job_satisfaction
sleep_amount          1.000000          0.684211
job_satisfaction      0.684211          1.000000

The Spearman's rank correlation coefficient is 0.68, it indicates a strong positive monotonic relationship. It means that as the amount of sleep individuals get each night increases, their overall job satisfaction tends to increase as well.

Q3. Suppose you are conducting a study to examine the relationship between the number of hours of
exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables
for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation
between these two variables and compare the results.

In [24]:
import numpy as np
import scipy.stats as stats

# Create the example dataset
exercise_hours = [4, 5, 3, 2, 6] # List of exercise hours for each participant
bmi = [22.3, 24.5, 27.1, 30.2, 25.8] # List of BMI values for each participant

# Calculate the Pearson correlation coefficient
pearson_coef = np.corrcoef(exercise_hours, bmi)[0, 1]

# Calculate the Spearman's rank correlation coefficient
spearman_coef, _ = stats.spearmanr(exercise_hours, bmi)

# Print the correlation coefficients
print("Pearson correlation coefficient:", pearson_coef)
print("Spearman's rank correlation coefficient:", spearman_coef)

Pearson correlation coefficient: -0.6108584466187268
Spearman's rank correlation coefficient: -0.6


Comparing the Results:
- Pearson correlation coefficient: The Pearson correlation coefficient measures the linear relationship between two variables. It assumes a linear relationship between the variables and calculates how closely the relationship follows a straight line.

- Spearman's rank correlation coefficient: The Spearman's rank correlation coefficient measures the monotonic relationship between two variables. It does not assume linearity but rather the direction and strength of the monotonic relationship


Both these results suggests, their is neqgative releationship b/w these two factors. Hence one will increase with decresing of other and vice-versa


Q4. A researcher is interested in examining the relationship between the number of hours individuals
spend watching television per day and their level of physical activity. The researcher collected data on
both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between
these two variables.

In [25]:
import numpy as np

# Create the example dataset
tv_hours = [3, 4, 2, 5, 6] # List of TV hours for each participant
physical_activity = [2, 3, 4, 1, 2] # List of physical activity levels for each participant

# Calculate the Pearson correlation coefficient
pearson_coef = np.corrcoef(tv_hours, physical_activity)[0, 1]

# Print the Pearson correlation coefficient
print("Pearson correlation coefficient:", pearson_coef)

Pearson correlation coefficient: -0.6933752452815363


Pearson correlation coefficient is -0.69, it indicates a strong negative linear relationship. It means that as the number of hours individuals spend watching television per day increases, their level of physical activity tends to decrease.

Q5. A survey was conducted to examine the relationship between age and preference for a particular
brand of soft drink. The survey results are shown below:

Age(Years)    Preferred Drink
25            Coke
42            Pepsi
37            Mountain dew
19            Coke
31            Pepsi
28            Coke



To examine the relationship between age and preference for a particular brand of soft drink based on the provided survey results, we can calculate the Phi coefficient. The Phi coefficient is a measure of association for two categorical variables, such as age group and preferred drink.

We can create a contingency table to calculate the Phi coefficient:

```
Coke Pepsi Mountain Dew
-----------------------------------------
<20 1 0 0
20-29 2 0 0
30-39 1 1 0
40+ 0 1 1
-----------------------------------------
```

Using this contingency table, we can calculate the Phi coefficient using the following formula:

```
Phi = (ad - bc) / sqrt((a+b)(c+d)(a+c)(b+d))
```

where a, b, c, and d represent the counts in each cell of the contingency table.

Calculating the Phi coefficient:

```
a = 1 (Count of individuals aged <20 who prefer Coke)
b = 1 (Count of individuals aged <20 who prefer Pepsi)
c = 1 (Count of individuals aged 30-39 who prefer Coke)
d = 2 (Count of individuals aged 40+ who prefer Pepsi or Mountain Dew)

Phi = (1*2 - 1*1) / sqrt((1+1)*(1+1)*(1+1)*(2+1))
= 1 / sqrt(2*2*2*3)
= 1 / sqrt(24)
≈ 0.204
```

Interpreting the result:
The Phi coefficient ranges from -1 to 1, where -1 indicates a perfect negative association, 0 indicates no association, and 1 indicates a perfect positive association.

In this case, the Phi coefficient of approximately 0.204 suggests a weak positive association between age and preference for a particular brand of soft drink. However, since the coefficient is close to 0, we can conclude that there is no strong relationship between age and the preferred drink based on this survey.

Note: The Phi coefficient is appropriate for categorical variables with two levels. If there are more than two levels for either age or preferred drink, other measures like Cramer's V or contingency coefficients should be considered for a more comprehensive analysis.

Q6. A company is interested in examining the relationship between the number of sales calls made per day
and the number of sales made per week. The company collected data on both variables from a sample of
30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [26]:
import numpy as np

# Create the example dataset
sales_calls_per_day = [10, 8, 12, 9, 7] # List of sales calls per day for each sales representative
sales_per_week = [5, 4, 7, 6, 3] # List of sales per week for each sales representative

# Calculate the Pearson correlation coefficient
pearson_coef = np.corrcoef(sales_calls_per_day, sales_per_week)[0, 1]

# Print the Pearson correlation coefficient
print("Pearson correlation coefficient:", pearson_coef)

Pearson correlation coefficient: 0.9041944301794651


Pearson correlation coefficient is 0.90, it indicates a strong positive linear relationship. It means that as the number of sales calls made per day increases, the number of sales made per week tends to increase as well.